test/test_unistd.rs

```diff
@@ -104,6 +104,60 @@ mod linux_android {
     }
 }
 
+#[test]
+// `getgroups()` and `setgroups()` do not behave as expected on Apple platforms
+#[cfg(not(any(target_os = "ios", target_os = "macos")))]
+fn test_setgroups() {
+    // Skip this test when not run as root as `setgroups()` requires root.
+    if !Uid::current().is_root() {
+        return
```

Comment:
Until Cargo adds a way to skip tests, you should print something here to let the user know that the test was skipped.  Otherwise it will rot, because people won't realize that the test isn't run.  See tests/test_mount.rs.  Ditto for line 137.